src/ranch.erl

```diff
@@ -71,7 +71,7 @@ start_listener(Ref, Transport, TransOpts0, Protocol, ProtoOpts)
 		when is_atom(Transport), is_atom(Protocol) ->
 	TransOpts = normalize_opts(TransOpts0),
 	_ = code:ensure_loaded(Transport),
-	case erlang:function_exported(Transport, name, 0) of
+	case validate_transopts(TransOpts) andalso erlang:function_exported(Transport, name, 0) of
 		false ->
 			{error, badarg};
```

Comment:
We probably should make that error more obvious as well.

Reply:
Returns `{error, {badtransport, Transport}}` now.